src/@context/Web3.tsx

```diff
@@ -304,6 +309,17 @@ function Web3Provider({ children }: { children: ReactNode }): ReactElement {
     }
     await web3Modal.clearCachedProvider()
   }
+  // -----------------------------------
+  // Get valid Networks and set isSupportedOceanNetwork
+  // -----------------------------------
+
+  useEffect(() => {
+    if (appConfig.chainIdsSupported.includes(networkId)) {
```

Comment:
let's clean up this logic as it's so short, it shouldn't be confusing at all. When next developer reads this logic, it says "when current user's networkId is among appConfig.chainIdsSupported, then network is supported, but when accountId is present, network is not supported". See how hard this logic is to get here?

This effect should only do what the comment says it does, check current user's network against supported networks, set to `false` if not. That's it. All the rest tries to accommodate to use case somewhere deep down in a component so this should be handled there.

Reply:
Ok yeah, I get your point here. I was thinking that we don't want the title to be red when the user doesn't have a wallet connected but I've moved that logic into the component now and this effect is simpler. 

```Typescript
<NetworkName
            networkId={networkId}
            className={
              isSupportedOceanNetwork || !accountId
                ? styles.network
                : `${styles.network} ${styles.error}`
            }
          />
```

Reply:
let's also update the effect dependencies as we only want this to fire when `networkId` or `appConfig.chainIdsSupported` is changed